src/Diagnostics.RuntimeHost/Services/GithubClient/GithubClient.cs

```diff
@@ -178,6 +180,13 @@ public async Task CreateOrUpdateFiles(IEnumerable<CommitContent> commits, string
             await _octokitClient.Git.Reference.Update(_userName, _repoName, headRef, new ReferenceUpdate(commit.Sha));
         }
 
+        public async Task<GitHubCommit> GetCommitByPath(string filePath)
+        {
+            var request = new CommitRequest { Path = filePath, Sha = _branch };
+            var commitLists = await _octokitClient.Repository.Commit.GetAll(_userName, _repoName, request);
+            return commitLists != null ? commitLists[0] : null;
```

Comment:
nit: you'll get an OutOfRange exception if CommitLists is Empty.
instead of returning commitLists[0], you can return commitLists.FirstOrDefault()